src/lib/core/TLVCommon.h

```diff
@@ -0,0 +1,96 @@
+/*
+ *
+ *    Copyright (c) 2020-2023 Project CHIP Authors
+ *    Copyright (c) 2013-2017 Nest Labs, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file contains common headers and definitions for working with data encoded in CHIP TLV format.
```

Comment:
I believe "common" has a strong potential to become a grab-all of things, like "util" or "helpers" or similar naming and I would prefer long term to avoid such names.

This file currently seems to contain size estimates and some operator| (unclear how to call that usage ... ).

Instead, the operators should be moved where TLVElementType is defined or similar and the estimates should be in `TLVSizeEstimates.h` or similar.